_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+
+No longer do you need to know all of your column names up front! DuckDB can select and even modify columns based on regular expression pattern matching, `EXCLUDE` or `REPLACE` modifiers, and even lambda functions (see the [section on lambda functions below](#list-lambda-functions) for details!). 
+
+Let’s take a look at some facts gathered about the first season of Star Trek. Using DuckDB’s [`httpfs` extension](https://duckdb.org/docs/extensions/httpfs), we can query a csv dataset directly from GitHub. It has several columns so let’s `DESCRIBE` it.
+```sql
+install httpfs;
+load httpfs;
+CREATE TABLE trek_facts AS
+    FROM 'https://raw.githubusercontent.com/Alex-Monahan/example_datasets/main/Star_Trek-Season_1.csv';
+
+DESCRIBE trek_facts;
+```
+
+|               column_name               | column_type | null | key  | default | extra |
+|:---|:---|:---|:---|:---|:---|
+| season_num                              | BIGINT      | YES  | NULL | NULL    | NULL  |
+| episode_num                             | BIGINT      | YES  | NULL | NULL    | NULL  |
+| aired_date                              | DATE        | YES  | NULL | NULL    | NULL  |
+| cnt_kirk_hookups                        | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_downed_redshirts                    | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_aliens_almost_took_over_planet     | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_aliens_almost_took_over_enterprise | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_vulcan_nerve_pinch                  | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_warp_speed_orders                   | BIGINT      | YES  | NULL | NULL    | NULL  |
+| highest_warp_speed_issued               | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_hand_phasers_fired                 | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_ship_phasers_fired                 | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_ship_photon_torpedos_fired         | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_transporter_pax                     | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_damn_it_jim_quote                   | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_im_givin_her_all_shes_got_quote     | BIGINT      | YES  | NULL | NULL    | NULL  |
+| cnt_highly_illogical_quote              | BIGINT      | YES  | NULL | NULL    | NULL  |
+| bool_enterprise_saved_the_day           | BIGINT      | YES  | NULL | NULL    | NULL  |
+
+#### COLUMNS() with regular expressions
+
+The `COLUMNS` expression can accept a string parameter that is a regular expression and will return all column names that match the pattern. How did warp change over the first season? Let’s examine any column name that contains the word `warp`.
+
+```sql
+FROM trek_facts
```

Comment:
Readers may be surprised by the FROM-first syntax -- it is explained later in the blogpost but maybe avoid using it before it is explained in the "FROM first" section.

Reply:
Excellent catch... Fixed!